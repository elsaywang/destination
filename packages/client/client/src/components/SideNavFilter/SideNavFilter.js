import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Tab, TabList } from '@react/react-spectrum/TabList';
import styles from './SideNavFilter.css';

class SideNavFilter extends Component {
    state = {
        selectedIndex: 0,
    };

    componentDidUpdate(prevProps) {
        if (!this.props.isSearched && prevProps.isSearched) {
            this.setState({ selectedIndex: 0 });
        }
    }

    handleNavFilterChange = tabIndex => {
        const { filterOptions } = this.props;
        const { value } = filterOptions[tabIndex];
        this.setState({ selectedIndex: tabIndex }, () => this.props.onFilterTypeChange(value));
    };

    render() {
        const { filterOptions, filterType } = this.props;
        return (
            <div className={styles.filterType}>
                <TabList
                    orientation="vertical"
                    variant="compact"
                    quiet
                    value={this.state.value}
                    onSelect={this.handleNavFilterChange}>
                    {filterOptions.map(({ label, value }) => (
                        <Tab
                            data-test={`${value.toLowerCase()}-type-filter`}
                            selected={filterType === value}
                            key={value}
                            value={value}
                            className={styles.filterType}>
                            {label}
                        </Tab>
                    ))}
                </TabList>
            </div>
        );
    }
}

SideNavFilter.propTypes = {
    isSearched: PropTypes.bool,
    onFilterTypeChange: PropTypes.func,
    filterType: PropTypes.string,
    filterOptions: PropTypes.arrayOf(
        PropTypes.shape({
            value: PropTypes.string,
            label: PropTypes.string,
        }),
    ),
};

export default SideNavFilter;
