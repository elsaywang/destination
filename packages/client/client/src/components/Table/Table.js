import PropTypes from "prop-types";
import React, { Component } from "react";
import classNames from "classnames";
import { TableView } from "@react/react-spectrum/TableView";
import DataSource from "./DataSource";
import {
	defaultRowHeight,
	defaultMaxRows,
	defaultHeadHeight
} from "../../constants/rows";
import styles from "./Table.css";

/**
 * Table component that wraps the react-spectrum table components
 * (`TableView` and `TableViewDataSource`). More easily accepts props
 * for items, columns, table config, and event handlers.
 */
class Table extends Component {
	constructor({ items, onLoadMore }) {
		super();
		this.dataSource = new DataSource({ items, onLoadMore });
	}

	/**
	 * Dynamically set the height of the table's container to show up to a
	 * certain number of rows.
	 */
	getTableHeight({
		items,
		rowHeight = defaultRowHeight,
		maxRows = defaultMaxRows
	}) {
		const bodyHeight = Math.min(items.length, maxRows) * rowHeight;

		return `${defaultHeadHeight + bodyHeight}px`;
	}

	render() {
		const {
			items,
			columns,
			rowHeight,
			renderCell,
			onSelectionChange,
			dataTest,
			allowsSelection
		} = this.props;

		const height = this.getTableHeight({ items, rowHeight });
		return (
			<div
				data-test={dataTest}
				className={classNames("table-wrapper", styles.tableWrapper)}
				style={{ height }}>
				<TableView
					dataSource={this.dataSource}
					columns={columns}
					renderCell={renderCell}
					rowHeight={rowHeight}
					onSelectionChange={onSelectionChange}
					allowsSelection={allowsSelection}
				/>
			</div>
		);
	}
}

Table.propTypes = {
	items: PropTypes.array.isRequired,
	columns: PropTypes.array.isRequired,
	renderCell: PropTypes.func.isRequired,
	onLoadMore: PropTypes.func,
	rowHeight: PropTypes.number,
	selectedRowIndexes: PropTypes.array,
	isMaxSelectedRowsReached: PropTypes.bool,
	onSelectionChange: PropTypes.func,
	sortSearch: PropTypes.func,
	dataTest: PropTypes.string,
	allowsSelection: PropTypes.bool
};

export default Table;
