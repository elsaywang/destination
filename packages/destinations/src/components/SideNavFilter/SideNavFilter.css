:local(.filterType) {
    padding: 1rem 0 0 0;
    font-size: 12px;
}

:local(.link) {
    color: inherit;
    text-decoration: none;
}
