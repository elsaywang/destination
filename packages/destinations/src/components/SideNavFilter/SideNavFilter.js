import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { NavLink, withRouter } from 'react-router-dom';
import { Tab, TabList } from '@react/react-spectrum/TabList';
import styles from './SideNavFilter.css';

const SideNavFilter = ({ filterOptions, filterType, onFilterTypeChange, baseRoute }) => (
    <div className={styles.filterType} data-test="side-nav-filter">
        <TabList
            orientation="vertical"
            variant="compact"
            quiet
            selectedIndex={filterOptions.findIndex(
                ({ value, isDefault }) =>
                    (_.isEmpty(filterType) && isDefault) || value === filterType,
            )}
            onChange={tabIndex => onFilterTypeChange(filterOptions[tabIndex])}>
            {filterOptions.map(({ label, value, subroute }) => (
                <Tab
                    data-test={`${value.toLowerCase().replace(/\W/g, '-')}-type-filter`}
                    key={value}
                    value={value}
                    className={styles.filterType}>
                    <NavLink className={styles.link} to={baseRoute + subroute}>
                        {label}
                    </NavLink>
                </Tab>
            ))}
        </TabList>
    </div>
);

SideNavFilter.propTypes = {
    isSearched: PropTypes.bool,
    onFilterTypeChange: PropTypes.func,
    filterType: PropTypes.string,
    filterOptions: PropTypes.arrayOf(
        PropTypes.shape({
            value: PropTypes.string,
            label: PropTypes.string,
        }),
    ),
};

export default withRouter(SideNavFilter);
