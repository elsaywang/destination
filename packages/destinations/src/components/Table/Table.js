import React, { Component } from 'react';
import { TableView } from '@react/react-spectrum/TableView';
import ListDataSource from '@react/react-spectrum/ListDataSource';
import IndexPath from '../../utils/indexPath';
import PropTypes from 'prop-types';

class Table extends Component {
    constructor({ reachedEndOfRows, items = [] }) {
        super();

        this.dataSource = new class DataSource extends ListDataSource {
            load() {
                return items;
            }

            performLoadMore() {
                let totalItems = 0;
                for (let i = 0; i < this.getNumberOfSections(); i++) {
                    totalItems += this.getSectionLength(i);
                }

                reachedEndOfRows(totalItems);
            }
        }();
    }

    componentWillReceiveProps(nextProps) {
        const length = this.props.items.length;

        this.dataSource.insertItems(new IndexPath(0, length), nextProps.items.slice(length));
    }

    render() {
        const { dataTest, styles, height, dataSource, items, ...tableViewProps } = this.props;

        return (
            <div data-test={dataTest} className={styles} style={{ height: `${height}px` }}>
                <TableView dataSource={this.dataSource} {...tableViewProps} />
            </div>
        );
    }
}

Table.defaultProps = {
    items: [],
    rowHeight: 30, // chosen randomly, not sure what a sensible default should be
    allowsSelection: false,
    reachedEndOfRows: () => {
        /* no-op*/
    },
};

Table.protoTypes = {
    height: PropTypes.number,
    columns: PropTypes.array,
    rowHeight: PropTypes.number,
    renderCell: PropTypes.func.isRequired,
    onSelectionChange: PropTypes.func,
    dataTest: PropTypes.string,
    allowsSelection: PropTypes.bool,
    styles: PropTypes.string,
    setRefreshDataSource: PropTypes.func,
    reachedEndOfRows: PropTypes.func,
};

export default Table;
