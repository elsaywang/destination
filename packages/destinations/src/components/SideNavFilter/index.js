import SideNavFilter from './SideNavFilter';

export default SideNavFilter;
