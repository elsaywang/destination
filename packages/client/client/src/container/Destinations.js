import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Table from '../components/Table';
import { columns, generateItems } from '../constants/tableData';
import SideNavFilter from '../components/SideNavFilter';
import Search from '@react/react-spectrum/Search';
import styles from './Destinations.css';
import { peopleBasedDestinationsTypeOptions } from '../constants/peopleBasedDestinationsOptions';

class Destinations extends Component {
    renderCell = (column, data) => {
        return <span>{data[column.key]}</span>;
    };

    showSideNavFilter = () => {
        const { destinationType } = this.props;
        return ['People-Based', 'Device-Based'].includes(destinationType);
    };

    componentWillMount() {
        this.setState({ items: this.props.items || generateItems() });
    }

    render() {
        const { destinationType, items } = this.props;
        const renderSideNavFilter = (
            <div className={styles.filterListContainer}>
                <SideNavFilter
                    onFilterTypeChange={() => {}}
                    filterType={'ALL'}
                    isSearched={true}
                    filterOptions={peopleBasedDestinationsTypeOptions}
                />
            </div>
        );
        return (
            <div
                style={{ display: 'flex' }}
                data-test={`${destinationType.toLowerCase()}-destinations`}>
                {this.showSideNavFilter() && renderSideNavFilter}
                <div className={styles.tableContainer}>
                    <div className={styles.search}>
                        <Search placeholder="Search" onChange={() => {}} onSubmit={() => {}} />
                    </div>
                    <Table
                        dataTest="peopleBased-destination-table"
                        items={this.state.items}
                        reachedEndOfRows={() => {
                            this.setState(prevState => {
                                return { items: prevState.items.concat(generateItems()) };
                            });
                        }}
                        height={500}
                        columns={columns}
                        rowHeight={500}
                        renderCell={this.renderCell}
                    />
                </div>
            </div>
        );
    }
}

Destinations.propTypes = {
    items: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.number,
            name: PropTypes.string,
            enabled: PropTypes.bool,
            createdBy: PropTypes.string,
        }),
    ),
    destinationType: PropTypes.string.isRequired,
};
export default Destinations;
