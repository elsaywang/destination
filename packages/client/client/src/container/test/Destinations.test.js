import React from 'react';
import { shallow } from 'enzyme';
import { Destinations } from '../Destinations';
import Table from '../../components/Table';
import SideNavFilter from '../../components/SideNavFilter';

describe('renders <App/> without crashing', () => {
    const props = {
        fetchDestinations: () => {
            /* no-op */
        },
        destinations: {
            list: [
                {
                    action: '',
                    description: 'Maxime ut totam nemo assumenda ratione impedit.',
                    id: 0,
                    name: 'input',
                    platform: 'Twitter',
                    shareableAudience: 41898,
                },
                {
                    action: '',
                    description: 'Et nesciunt voluptatibus facilis blanditiis.',
                    id: 1,
                    name: 'Bedfordshire',
                    platform: 'Google',
                    shareableAudience: 18751,
                },
                {
                    action: '',
                    description: 'Repellat sit dolor ut quisquam et ducimus.',
                    id: 2,
                    name: 'Shoes',
                    platform: 'Twitter',
                    shareableAudience: 5434,
                },
                {
                    action: '',
                    description: 'Blanditiis tenetur culpa aut distinctio.',
                    id: 3,
                    name: 'mesh',
                    platform: 'LinkedIn',
                    shareableAudience: 40918,
                },
            ],
            requestInFlight: false,
        },
        destinationType: 'People-Based',
    };
    const wrapper = shallow(<Destinations {...props} />);
    describe('rendering', () => {
        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });
        it('renders <Table/>', () => {
            expect(wrapper.find(Table).exists()).toBeTruthy();
        });
        it('renders <SideNavFilter/> when `destinationType` is `People-Based`', () => {
            expect(wrapper.find(SideNavFilter).exists()).toBeTruthy();
        });
        it('renders no <SideNavFilter/> when `destinationType` is non-`People-Based`', () => {
            wrapper.setProps({ destinationType: 'Cookie-Based' });
            expect(wrapper.find(SideNavFilter).exists()).toBeFalsy();
        });
    });
});
