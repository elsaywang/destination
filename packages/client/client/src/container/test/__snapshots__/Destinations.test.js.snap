// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <App/> without crashing rendering matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Destinations
    destinationType="People-Based"
    items={
        Array [
            Object {
              "action": "",
              "description": "Maxime ut totam nemo assumenda ratione impedit.",
              "id": 0,
              "name": "input",
              "platform": "Twitter",
              "shareableAudience": 41898,
            },
            Object {
              "action": "",
              "description": "Et nesciunt voluptatibus facilis blanditiis.",
              "id": 1,
              "name": "Bedfordshire",
              "platform": "Google",
              "shareableAudience": 18751,
            },
            Object {
              "action": "",
              "description": "Repellat sit dolor ut quisquam et ducimus.",
              "id": 2,
              "name": "Shoes",
              "platform": "Twitter",
              "shareableAudience": 5434,
            },
            Object {
              "action": "",
              "description": "Blanditiis tenetur culpa aut distinctio.",
              "id": 3,
              "name": "mesh",
              "platform": "LinkedIn",
              "shareableAudience": 40918,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className={undefined}
>
          <SideNavFilter
                    filterOptions={
                              Array [
                                        Object {
                                          "label": "ALL",
                                          "value": "ALL",
                                        },
                                        Object {
                                          "label": "FACEBOOK",
                                          "value": "FB",
                                        },
                                        Object {
                                          "label": "GOOGLE",
                                          "value": "GOOGL",
                                        },
                                        Object {
                                          "label": "LINKEDIN",
                                          "value": "LNKD",
                                        },
                                        Object {
                                          "label": "TWITTER",
                                          "value": "TWTR",
                                        },
                                      ]
                    }
                    filterType="ALL"
                    isSearched={true}
                    onFilterTypeChange={[Function]}
          />
</div>,
        <div
          className={undefined}
>
          <div
                    className={undefined}
          >
                    <Search
                              icon={<Magnifier />}
                              onChange={[Function]}
                              onSubmit={[Function]}
                              placeholder="Search"
                    />
          </div>
          <Table
                    allowsSelection={false}
                    columns={
                              Array [
                                        Object {
                                          "announce": false,
                                          "key": "id",
                                          "sortable": true,
                                          "title": "ID",
                                          "width": 50,
                                        },
                                        Object {
                                          "active": true,
                                          "key": "platform",
                                          "title": "PLATFORM",
                                          "width": 150,
                                        },
                                        Object {
                                          "key": "name",
                                          "title": "NAME",
                                          "width": 200,
                                        },
                                        Object {
                                          "key": "description",
                                          "minWidth": 200,
                                          "title": "DESCRIPTION",
                                        },
                                        Object {
                                          "key": "shareableAudience",
                                          "sortable": true,
                                          "title": "SHAREABLE AUDIENCE",
                                          "width": 100,
                                        },
                                        Object {
                                          "key": "action",
                                          "title": "ACTION",
                                          "width": 150,
                                        },
                                      ]
                    }
                    dataTest="peopleBased-destination-table"
                    height={500}
                    items={
                              Array [
                                        Object {
                                          "action": "",
                                          "description": "Maxime ut totam nemo assumenda ratione impedit.",
                                          "id": 0,
                                          "name": "input",
                                          "platform": "Twitter",
                                          "shareableAudience": 41898,
                                        },
                                        Object {
                                          "action": "",
                                          "description": "Et nesciunt voluptatibus facilis blanditiis.",
                                          "id": 1,
                                          "name": "Bedfordshire",
                                          "platform": "Google",
                                          "shareableAudience": 18751,
                                        },
                                        Object {
                                          "action": "",
                                          "description": "Repellat sit dolor ut quisquam et ducimus.",
                                          "id": 2,
                                          "name": "Shoes",
                                          "platform": "Twitter",
                                          "shareableAudience": 5434,
                                        },
                                        Object {
                                          "action": "",
                                          "description": "Blanditiis tenetur culpa aut distinctio.",
                                          "id": 3,
                                          "name": "mesh",
                                          "platform": "LinkedIn",
                                          "shareableAudience": 40918,
                                        },
                                      ]
                    }
                    reachedEndOfRows={[Function]}
                    renderCell={[Function]}
                    rowHeight={500}
          />
</div>,
      ],
      "data-test": "people-based-destinations",
      "style": Object {
        "display": "flex",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <SideNavFilter
            filterOptions={
                        Array [
                                    Object {
                                      "label": "ALL",
                                      "value": "ALL",
                                    },
                                    Object {
                                      "label": "FACEBOOK",
                                      "value": "FB",
                                    },
                                    Object {
                                      "label": "GOOGLE",
                                      "value": "GOOGL",
                                    },
                                    Object {
                                      "label": "LINKEDIN",
                                      "value": "LNKD",
                                    },
                                    Object {
                                      "label": "TWITTER",
                                      "value": "TWTR",
                                    },
                                  ]
            }
            filterType="ALL"
            isSearched={true}
            onFilterTypeChange={[Function]}
/>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "filterOptions": Array [
              Object {
                "label": "ALL",
                "value": "ALL",
              },
              Object {
                "label": "FACEBOOK",
                "value": "FB",
              },
              Object {
                "label": "GOOGLE",
                "value": "GOOGL",
              },
              Object {
                "label": "LINKEDIN",
                "value": "LNKD",
              },
              Object {
                "label": "TWITTER",
                "value": "TWTR",
              },
            ],
            "filterType": "ALL",
            "isSearched": true,
            "onFilterTypeChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className={undefined}
>
              <Search
                            icon={<Magnifier />}
                            onChange={[Function]}
                            onSubmit={[Function]}
                            placeholder="Search"
              />
</div>,
            <Table
              allowsSelection={false}
              columns={
                            Array [
                                          Object {
                                            "announce": false,
                                            "key": "id",
                                            "sortable": true,
                                            "title": "ID",
                                            "width": 50,
                                          },
                                          Object {
                                            "active": true,
                                            "key": "platform",
                                            "title": "PLATFORM",
                                            "width": 150,
                                          },
                                          Object {
                                            "key": "name",
                                            "title": "NAME",
                                            "width": 200,
                                          },
                                          Object {
                                            "key": "description",
                                            "minWidth": 200,
                                            "title": "DESCRIPTION",
                                          },
                                          Object {
                                            "key": "shareableAudience",
                                            "sortable": true,
                                            "title": "SHAREABLE AUDIENCE",
                                            "width": 100,
                                          },
                                          Object {
                                            "key": "action",
                                            "title": "ACTION",
                                            "width": 150,
                                          },
                                        ]
              }
              dataTest="peopleBased-destination-table"
              height={500}
              items={
                            Array [
                                          Object {
                                            "action": "",
                                            "description": "Maxime ut totam nemo assumenda ratione impedit.",
                                            "id": 0,
                                            "name": "input",
                                            "platform": "Twitter",
                                            "shareableAudience": 41898,
                                          },
                                          Object {
                                            "action": "",
                                            "description": "Et nesciunt voluptatibus facilis blanditiis.",
                                            "id": 1,
                                            "name": "Bedfordshire",
                                            "platform": "Google",
                                            "shareableAudience": 18751,
                                          },
                                          Object {
                                            "action": "",
                                            "description": "Repellat sit dolor ut quisquam et ducimus.",
                                            "id": 2,
                                            "name": "Shoes",
                                            "platform": "Twitter",
                                            "shareableAudience": 5434,
                                          },
                                          Object {
                                            "action": "",
                                            "description": "Blanditiis tenetur culpa aut distinctio.",
                                            "id": 3,
                                            "name": "mesh",
                                            "platform": "LinkedIn",
                                            "shareableAudience": 40918,
                                          },
                                        ]
              }
              reachedEndOfRows={[Function]}
              renderCell={[Function]}
              rowHeight={500}
/>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Search
                icon={<Magnifier />}
                onChange={[Function]}
                onSubmit={[Function]}
                placeholder="Search"
/>,
              "className": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "icon": <Magnifier />,
                "onChange": [Function],
                "onSubmit": [Function],
                "placeholder": "Search",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowsSelection": false,
              "columns": Array [
                Object {
                  "announce": false,
                  "key": "id",
                  "sortable": true,
                  "title": "ID",
                  "width": 50,
                },
                Object {
                  "active": true,
                  "key": "platform",
                  "title": "PLATFORM",
                  "width": 150,
                },
                Object {
                  "key": "name",
                  "title": "NAME",
                  "width": 200,
                },
                Object {
                  "key": "description",
                  "minWidth": 200,
                  "title": "DESCRIPTION",
                },
                Object {
                  "key": "shareableAudience",
                  "sortable": true,
                  "title": "SHAREABLE AUDIENCE",
                  "width": 100,
                },
                Object {
                  "key": "action",
                  "title": "ACTION",
                  "width": 150,
                },
              ],
              "dataTest": "peopleBased-destination-table",
              "height": 500,
              "items": Array [
                Object {
                  "action": "",
                  "description": "Maxime ut totam nemo assumenda ratione impedit.",
                  "id": 0,
                  "name": "input",
                  "platform": "Twitter",
                  "shareableAudience": 41898,
                },
                Object {
                  "action": "",
                  "description": "Et nesciunt voluptatibus facilis blanditiis.",
                  "id": 1,
                  "name": "Bedfordshire",
                  "platform": "Google",
                  "shareableAudience": 18751,
                },
                Object {
                  "action": "",
                  "description": "Repellat sit dolor ut quisquam et ducimus.",
                  "id": 2,
                  "name": "Shoes",
                  "platform": "Twitter",
                  "shareableAudience": 5434,
                },
                Object {
                  "action": "",
                  "description": "Blanditiis tenetur culpa aut distinctio.",
                  "id": 3,
                  "name": "mesh",
                  "platform": "LinkedIn",
                  "shareableAudience": 40918,
                },
              ],
              "reachedEndOfRows": [Function],
              "renderCell": [Function],
              "rowHeight": 500,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={undefined}
>
            <SideNavFilter
                        filterOptions={
                                    Array [
                                                Object {
                                                  "label": "ALL",
                                                  "value": "ALL",
                                                },
                                                Object {
                                                  "label": "FACEBOOK",
                                                  "value": "FB",
                                                },
                                                Object {
                                                  "label": "GOOGLE",
                                                  "value": "GOOGL",
                                                },
                                                Object {
                                                  "label": "LINKEDIN",
                                                  "value": "LNKD",
                                                },
                                                Object {
                                                  "label": "TWITTER",
                                                  "value": "TWTR",
                                                },
                                              ]
                        }
                        filterType="ALL"
                        isSearched={true}
                        onFilterTypeChange={[Function]}
            />
</div>,
          <div
            className={undefined}
>
            <div
                        className={undefined}
            >
                        <Search
                                    icon={<Magnifier />}
                                    onChange={[Function]}
                                    onSubmit={[Function]}
                                    placeholder="Search"
                        />
            </div>
            <Table
                        allowsSelection={false}
                        columns={
                                    Array [
                                                Object {
                                                  "announce": false,
                                                  "key": "id",
                                                  "sortable": true,
                                                  "title": "ID",
                                                  "width": 50,
                                                },
                                                Object {
                                                  "active": true,
                                                  "key": "platform",
                                                  "title": "PLATFORM",
                                                  "width": 150,
                                                },
                                                Object {
                                                  "key": "name",
                                                  "title": "NAME",
                                                  "width": 200,
                                                },
                                                Object {
                                                  "key": "description",
                                                  "minWidth": 200,
                                                  "title": "DESCRIPTION",
                                                },
                                                Object {
                                                  "key": "shareableAudience",
                                                  "sortable": true,
                                                  "title": "SHAREABLE AUDIENCE",
                                                  "width": 100,
                                                },
                                                Object {
                                                  "key": "action",
                                                  "title": "ACTION",
                                                  "width": 150,
                                                },
                                              ]
                        }
                        dataTest="peopleBased-destination-table"
                        height={500}
                        items={
                                    Array [
                                                Object {
                                                  "action": "",
                                                  "description": "Maxime ut totam nemo assumenda ratione impedit.",
                                                  "id": 0,
                                                  "name": "input",
                                                  "platform": "Twitter",
                                                  "shareableAudience": 41898,
                                                },
                                                Object {
                                                  "action": "",
                                                  "description": "Et nesciunt voluptatibus facilis blanditiis.",
                                                  "id": 1,
                                                  "name": "Bedfordshire",
                                                  "platform": "Google",
                                                  "shareableAudience": 18751,
                                                },
                                                Object {
                                                  "action": "",
                                                  "description": "Repellat sit dolor ut quisquam et ducimus.",
                                                  "id": 2,
                                                  "name": "Shoes",
                                                  "platform": "Twitter",
                                                  "shareableAudience": 5434,
                                                },
                                                Object {
                                                  "action": "",
                                                  "description": "Blanditiis tenetur culpa aut distinctio.",
                                                  "id": 3,
                                                  "name": "mesh",
                                                  "platform": "LinkedIn",
                                                  "shareableAudience": 40918,
                                                },
                                              ]
                        }
                        reachedEndOfRows={[Function]}
                        renderCell={[Function]}
                        rowHeight={500}
            />
</div>,
        ],
        "data-test": "people-based-destinations",
        "style": Object {
          "display": "flex",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <SideNavFilter
              filterOptions={
                            Array [
                                          Object {
                                            "label": "ALL",
                                            "value": "ALL",
                                          },
                                          Object {
                                            "label": "FACEBOOK",
                                            "value": "FB",
                                          },
                                          Object {
                                            "label": "GOOGLE",
                                            "value": "GOOGL",
                                          },
                                          Object {
                                            "label": "LINKEDIN",
                                            "value": "LNKD",
                                          },
                                          Object {
                                            "label": "TWITTER",
                                            "value": "TWTR",
                                          },
                                        ]
              }
              filterType="ALL"
              isSearched={true}
              onFilterTypeChange={[Function]}
/>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "filterOptions": Array [
                Object {
                  "label": "ALL",
                  "value": "ALL",
                },
                Object {
                  "label": "FACEBOOK",
                  "value": "FB",
                },
                Object {
                  "label": "GOOGLE",
                  "value": "GOOGL",
                },
                Object {
                  "label": "LINKEDIN",
                  "value": "LNKD",
                },
                Object {
                  "label": "TWITTER",
                  "value": "TWTR",
                },
              ],
              "filterType": "ALL",
              "isSearched": true,
              "onFilterTypeChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className={undefined}
>
                <Search
                                icon={<Magnifier />}
                                onChange={[Function]}
                                onSubmit={[Function]}
                                placeholder="Search"
                />
</div>,
              <Table
                allowsSelection={false}
                columns={
                                Array [
                                                Object {
                                                  "announce": false,
                                                  "key": "id",
                                                  "sortable": true,
                                                  "title": "ID",
                                                  "width": 50,
                                                },
                                                Object {
                                                  "active": true,
                                                  "key": "platform",
                                                  "title": "PLATFORM",
                                                  "width": 150,
                                                },
                                                Object {
                                                  "key": "name",
                                                  "title": "NAME",
                                                  "width": 200,
                                                },
                                                Object {
                                                  "key": "description",
                                                  "minWidth": 200,
                                                  "title": "DESCRIPTION",
                                                },
                                                Object {
                                                  "key": "shareableAudience",
                                                  "sortable": true,
                                                  "title": "SHAREABLE AUDIENCE",
                                                  "width": 100,
                                                },
                                                Object {
                                                  "key": "action",
                                                  "title": "ACTION",
                                                  "width": 150,
                                                },
                                              ]
                }
                dataTest="peopleBased-destination-table"
                height={500}
                items={
                                Array [
                                                Object {
                                                  "action": "",
                                                  "description": "Maxime ut totam nemo assumenda ratione impedit.",
                                                  "id": 0,
                                                  "name": "input",
                                                  "platform": "Twitter",
                                                  "shareableAudience": 41898,
                                                },
                                                Object {
                                                  "action": "",
                                                  "description": "Et nesciunt voluptatibus facilis blanditiis.",
                                                  "id": 1,
                                                  "name": "Bedfordshire",
                                                  "platform": "Google",
                                                  "shareableAudience": 18751,
                                                },
                                                Object {
                                                  "action": "",
                                                  "description": "Repellat sit dolor ut quisquam et ducimus.",
                                                  "id": 2,
                                                  "name": "Shoes",
                                                  "platform": "Twitter",
                                                  "shareableAudience": 5434,
                                                },
                                                Object {
                                                  "action": "",
                                                  "description": "Blanditiis tenetur culpa aut distinctio.",
                                                  "id": 3,
                                                  "name": "mesh",
                                                  "platform": "LinkedIn",
                                                  "shareableAudience": 40918,
                                                },
                                              ]
                }
                reachedEndOfRows={[Function]}
                renderCell={[Function]}
                rowHeight={500}
/>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Search
                  icon={<Magnifier />}
                  onChange={[Function]}
                  onSubmit={[Function]}
                  placeholder="Search"
/>,
                "className": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "icon": <Magnifier />,
                  "onChange": [Function],
                  "onSubmit": [Function],
                  "placeholder": "Search",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "allowsSelection": false,
                "columns": Array [
                  Object {
                    "announce": false,
                    "key": "id",
                    "sortable": true,
                    "title": "ID",
                    "width": 50,
                  },
                  Object {
                    "active": true,
                    "key": "platform",
                    "title": "PLATFORM",
                    "width": 150,
                  },
                  Object {
                    "key": "name",
                    "title": "NAME",
                    "width": 200,
                  },
                  Object {
                    "key": "description",
                    "minWidth": 200,
                    "title": "DESCRIPTION",
                  },
                  Object {
                    "key": "shareableAudience",
                    "sortable": true,
                    "title": "SHAREABLE AUDIENCE",
                    "width": 100,
                  },
                  Object {
                    "key": "action",
                    "title": "ACTION",
                    "width": 150,
                  },
                ],
                "dataTest": "peopleBased-destination-table",
                "height": 500,
                "items": Array [
                  Object {
                    "action": "",
                    "description": "Maxime ut totam nemo assumenda ratione impedit.",
                    "id": 0,
                    "name": "input",
                    "platform": "Twitter",
                    "shareableAudience": 41898,
                  },
                  Object {
                    "action": "",
                    "description": "Et nesciunt voluptatibus facilis blanditiis.",
                    "id": 1,
                    "name": "Bedfordshire",
                    "platform": "Google",
                    "shareableAudience": 18751,
                  },
                  Object {
                    "action": "",
                    "description": "Repellat sit dolor ut quisquam et ducimus.",
                    "id": 2,
                    "name": "Shoes",
                    "platform": "Twitter",
                    "shareableAudience": 5434,
                  },
                  Object {
                    "action": "",
                    "description": "Blanditiis tenetur culpa aut distinctio.",
                    "id": 3,
                    "name": "mesh",
                    "platform": "LinkedIn",
                    "shareableAudience": 40918,
                  },
                ],
                "reachedEndOfRows": [Function],
                "renderCell": [Function],
                "rowHeight": 500,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
